src/FirmwarePlugin/APM/APMFirmwarePlugin.cc

```diff
@@ -167,5 +251,66 @@ void APMFirmwarePlugin::adjustMavlinkMessage(mavlink_message_t* message)
         mavlink_msg_param_set_encode(message->sysid, message->compid, message, &paramSet);
     }
 
-    // FIXME: Need to implement mavlink message severity adjustment
+    if (message->msgid == MAVLINK_MSG_ID_STATUSTEXT)
+    {
+        QByteArray b;
```

Comment:
Could you put an if (!_firmwareVersion.isValid()) around the version parsing code so that it stops parsing after it has found the version? Trying to save a little cpu.